File: 02 백준/10 문자열/Python/Prob9342.py

"""
title  : 9342. 염색체 (Silver 3)
time   : 108ms        180ms
memory : 34752KB      112896KB
"""

import sys
import re

def solution():

    input = sys.stdin.readline
    T = int(input())    # 테스트케이스 개수

    # 정규표현식
    check = re.compile('[ABCDEF]?A+F+C+[ABCDEF]?$')

    for _ in range(T):       
        print("Infected!" if check.match(input().rstrip()) else "Good")

solution()

File: 02 백준/10 문자열/Java/src/Prob9342.java

/**
* title  : 9342. 염색체 (Silver 3)
* time   : 108ms
* memory : 14176KB
*/

import java.io.*;

public class Prob9342 {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        String pattern = "^[ABCDEF]?A+F+C+[ABCDEF]?$";

        int N = Integer.parseInt(br.readLine());
        for (int i = 0; i < N; i++) {

            String word = br.readLine();
            
            if (word.matches(pattern)) {
                sb.append("Infected!").append("\n");
            } else {
                sb.append("Good").append("\n");
            }
        }
        
        System.out.println(sb);
    }
}